File: angularapp/src/app/services/vehicle.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { VehicleMaintenance } from '../models/vehicle-maintenance.model';
import { API_URL } from '../constants/app.constant';

@Injectable({
  providedIn: 'root'
})
export class VehicleService {

   constructor(private http:HttpClient) { }

    public getAllServices():Observable<VehicleMaintenance[]>
    {
      return this.http.get<VehicleMaintenance[]>(API_URL+"/services");
    }

    public addService(service:VehicleMaintenance):Observable<VehicleMaintenance>
    {
      return this.http.post<VehicleMaintenance>(API_URL+"/services",service);
    }

    public updateService(serviceId:number,updatedService:VehicleMaintenance):Observable<VehicleMaintenance>
    {
      return this.http.put<VehicleMaintenance>(API_URL+"/services/"+serviceId,updatedService);
    }

    public deleteService(serviceId:Number):Observable<any>
    {
      return this.http.delete<any>(API_URL+"/services/"+serviceId);
    }

    public getServiceByName(serviceName:string):Observable<VehicleMaintenance[]>
    {
      return this.http.get<VehicleMaintenance[]>(API_URL+"/services"+serviceName);
    }

    public getServiceById(serviceId:number):Observable<VehicleMaintenance>
    {
      return this.http.get<VehicleMaintenance>(API_URL+"/services/"+serviceId);
    }

    
  public checkPendingAppointments(serviceId: number): Observable<boolean> {
    return this.http.get<boolean>(API_URL + "/appointment/services/" + serviceId + "/hasPending");
  }

  public updateServiceStatus(serviceId: number, status: string): Observable<void> {
    return this.http.put<void>(API_URL + "/services/" + serviceId + "/status?status=" + status, {});
  }

    

}

File: springapp/src/main/java/com/examly/springapp/service/VehicleService.java
package com.examly.springapp.service;

import java.util.List;
import java.util.Optional;


import com.examly.springapp.dto.VehicleMaintenanceDto;



public interface VehicleService {
    public VehicleMaintenanceDto addService(VehicleMaintenanceDto serviceDto);
    public VehicleMaintenanceDto updateService(Long serviceId,VehicleMaintenanceDto serviceDto);
    public void deleteService(Long serviceId);
    public List<VehicleMaintenanceDto> getAllServices();
    public Optional<VehicleMaintenanceDto> getServiceById(Long serviceId);
    public List<VehicleMaintenanceDto> findByServiceName(String serviceName);
    public void updateStatus(Long serviceId, String status);

   
    
}
